pollux/vc-jwt/src/main/scala/org/hyperledger/identus/pollux/vc/jwt/JWTVerification.scala

```diff
@@ -125,6 +137,61 @@ object JWTVerification {
       })
   }
 
+  def keyIdDIDExtractor(jwt: JWT): Validation[String, String] = {
+    for {
+      header <- Validation
+        .fromTry(
+          JwtCirce
+            .decodeAll(jwt.value, JwtOptions(false, false, false))
+            .map(_._1)
+        )
+        .mapError(_.getMessage)
+      keyId <- Validation.fromOptionWith("Key ID not found in JWT header")(header.keyId)
+      isValidPrismDID <- Validation.fromEither(PrismDID.fromString(keyId)) // TODO: we don't support other DIDs yet
+    } yield keyId
+  }
+
+  def validateEncodedJwtWithKeyId(
+      jwt: JWT,
+      proofPurpose: Option[VerificationRelationship] = None,
+      didResolver: DidResolver,
+      didExtractor: JWT => Validation[String, String] = keyIdDIDExtractor
+  ): IO[String, Validation[String, Unit]] = {
+    val decodedJWT = Validation
+      .fromTry(JwtCirce.decodeRawAll(jwt.value, JwtOptions(false, false, false)))
+      .mapError(_.getMessage)
+
+    val extractAlgorithm: Validation[String, JwtAlgorithm] =
+      for {
+        decodedJwtTask <- decodedJWT
+        (header, _, _) = decodedJwtTask
+        algorithm <- Validation
+          .fromOptionWith("An algorithm must be specified in the header")(JwtCirce.parseHeader(header).algorithm)
+      } yield algorithm
+
+    val claim: Validation[String, String] =
+      for {
+        decodedJwtTask <- decodedJWT
+        (_, claim, _) = decodedJwtTask
+      } yield claim
+
+    val extractedDID = didExtractor(jwt)
```

Comment:
Should this instead extract DID and fragment (key-id), which then used when we `extractVerificationMethods`?

Reply:
you are right, I will refactor this later.
right now I don't pass the #fragment with key-id and I'm not sure that we support key-id in the `extractVerificationMethods` as we support `VerificationRelationship` only :(